src/portable/nxp/lpc_ip3511/dcd_lpc_ip3511.c

```diff
@@ -174,6 +178,7 @@ typedef struct
 //    For example: LPC55s69 port1 Highspeed must be USB_RAM (0x40100000)
 //    Use CFG_TUSB_MEM_SECTION to place it accordingly.
 CFG_TUSB_MEM_SECTION TU_ATTR_ALIGNED(256) static dcd_data_t _dcd;
+CFG_TUSB_MEM_SECTION TU_ATTR_ALIGNED(256) static volatile uint8_t dummy[64]; // TODO temp fix to prevent accidental overwrite due to ep[][].buffer_xx.offset being 0
```

Comment:
do we really need to have this. Have you experience issue with dcd data is overwritten. USB data shouldn't be accepted when active bit is not set

Reply:
I think there is likely a better solution, but it was causing me some issues not assigning the offsets to another location.

In functions such as ```dcd_init()``` and others, the USB would write to the offset in ```_dcd.ep[...][...].buffer_xx.offset``` and I think because the value of the ```DATABUFSTART``` register is 0x2000000, the USB was overwriting some other values I had in SRAM. The reason I noticed it was because the LED ```blink_interval_ms``` in ```main()``` was not changing correctly and it happened to be placed at 0x20000000 in my build. First place it happened was  after ```dcd_reg->DEVCMDSTAT  |= ...``` in ```dcd_init()```

Reply:
it is not clear enough that USB overwrite the data, even if that is the case, I think we should properly fix it e.g explicitly set SKIP/DISABLE bit for endpoint first before zero it out. Otherwise we could run into issue that host may think we already received data (overwritten to zero), while it is ended up in dummy. Maybe we should do it in a separated PR.

Reply:
I think I noticed it on the video example when I was testing because ```blink_interval_ms``` was getting overwritten at address 0x20000000. I will take a look again as well as seeing if I can find proper solution.

Reply:
A repeatable setup for me:
- example: video_capture
- modifications: revert all lines using ```dummy``` to before
- OS: Windows
- when USB is asleep it blinks find. open the camera app, once the device is awake, the value of ```blink_interval_ms``` is 0 (address 0x20000000)

Reply:
I think I have narrowed it down to ```dcd_edpt_xfer()``` when ```buffer``` is ```NULL```, occurring for EP0 OUT ZLPs. The code below makes it modify ```dummy```, and the other previous locations of ```dummy``` can be removed. From what I have seen, the USB writes 4 bytes of zeros to its offset location in this scenario. I am not sure if there is a proper solution to this aside from a dummy buffer?
```
bool dcd_edpt_xfer(uint8_t rhport, uint8_t ep_addr, uint8_t* buffer, uint16_t total_bytes)
{
  uint8_t const ep_id = ep_addr2id(ep_addr);

  tu_memclr(&_dcd.dma[ep_id], sizeof(xfer_dma_t));
  _dcd.dma[ep_id].total_bytes = total_bytes;

  if (!buffer && !ep_id)
  {
    buffer = (uint8_t*)(uint32_t)dummy;
  }
  prepare_ep_xfer(rhport, ep_id, get_buf_offset(buffer), total_bytes);

  return true;
}
```